process/project_milestones.md

```diff
@@ -0,0 +1,51 @@
+# Cloud Native Milestones
+
+Milestones are intended to provide checkpoints for projects as they grow and mature in the cloud native ecosystem. When leveraged, projects may find their path towards Graduation is more structured, clear, and allows them to reach maturity in a more robust and well understood manner. Milestones do not guarantee a project’s graduation within the CNCF, rather, they are a collection of key activities that have helped other projects achieve maturity, stability, and adoption. Even after projects are graduated, they may leverage milestones as a mechanism to continue sustaining their project in a manner expected by their community and adopters particularly as they experience change and turnover. The community is welcome to submit PRs to improve and add additional milestones based on their experiences in maturing their projects.
+
+## Sandbox
+
+Some milestones may not apply to all projects and should be leveraged as guiding points. Milestones are described as the desired state with examples for how to achieve this.
+* Project has the basics of security underway
+  * Completion of the [self-assessment](https://github.com/cncf/tag-security/blob/main/assessments/guide/self-assessment.md)
+* Project holds community meetings or active discussions
+  * Zoom with meeting notes occurring more than twice a quarter (or community.cncf.io!) 
+  * Use of GitHub discussions, issues, or a group messaging service (slack, wechat, etc.)
+* Project guides community contributions
+  * PRs from community members receive timely feedback and discussion from the maintainers
+* Project works with an angel adopter
+  * Partners closely with the project to address organizational needs of the project early on
+* Project begins solidifying a versioning schema and release cadence
+  * A roadmap clearly defines initiatives that align with releases
+  * Releases begin to have establish a pattern of regularity
+* Project has engaged a Technical Advisory Group for feedback on some portion of the project relevant to the domain
+* Project has a clearly discoverable governance doc that covers the basics of decision making to include the commit bit
```

Comment:
```suggestion
* Project has a clearly discoverable governance doc that covers the basics of decision making and how to earn the commit bit
```

Reply:
"the commit bit" is probably not a well understood term.  I don't know if the CNCF has a defined term (["maintainer" is definitely overloaded](https://github.com/cncf/foundation/issues/433)), but I think "how to earn permission to approve pull requests" or something similar would help.

(I assume the SCM for CNCF projects does not _have_ to include GitHub and a more inclusive terms than Pull Requests might be appropriate)

Reply:
I've been thinking about Roadmaps. In the past I've been in favour of them as a requirement but I think I'm changing my mind because I'm not convinced they are as valuable as we'd like to think. 

Background / food for thought: the Linux kernel does not have a published roadmap. Everyone involved knows what's going through mailing list discussions.

When we say things like a roadmap that's aligned with releases, combined with a regular cadence of releases, this feels dangerously close to requiring open source projects to make commitments that end users might come to think they can rely on.

The main concern I have is that it doesn't seem very aligned with the ethos of open source to require developers to sign up to fixed commitments. Sure, in our community most people are being paid to work on projects but not everyone is. And if someone from Company A makes a commitment to deliver a particular feature in a certain release, but then Company A decides they should spend their time elsewhere, who from the project is supposed to pick up the slack to make sure the commitment is reached? 

It might work for very large projects, but the more I think about it, the more it seems like a recipe for burnout amongst contributors, and disappointment amongst end users. 

Projects generally use issues to communicate what features and fixes people are working on. A feature that meets some set of criteria (up to the project) makes it into the next release. Isn't that sufficient? 

And if we require more, isn't that rather overstepping the idea that we don't impose any particular governance on projects?

Reply:
+100 - we really need to stop turning CNCF into a giant bureaucracy that treats all projects the same.  This kills innovation stone dead and will put so many people off.  Also, please stop imposing requirements on project maintainers that add a lot of work and will lead to burnout, without a very strong compensating reason and balance

Reply:
@lizrice Thanks for bringing this up, I dont disagree. We currently have roadmaps integrated through various portions of the lifecycle so we'll need to evaluate what the value is of these in both understanding the project's longer goals as well as the direction they are heading, however they are always point in time for evaluations along the lifecycle. 

Your point on perceived commitment is spot on as well, and I'm not confident "messaging" here will resolve this disconnect. We'll need to consider more about how we can view the direction of the project at an evaluation, without dictating how the project expresses that.

Reply:
@monadic Thank you for the feedback.  This PR on milestones does not establish new or additional requirements on projects, it is designed to serve as ` guiding points along their [project's] journey — from sandbox to beyond graduation` which they `may leverage`.  However, as @craigbox mentioned later, these may be seen as requirements. This is something will need to rework in this PR to avoid.
Regarding burnout, agreed. The CNCF lifecycle is structured to evaluate projects for stability, maturity, and adoption potential (to varying degrees) over the years, as we've learned more about project success, we're beginning to distill steps projects have taken along their path to success. Not as requirements, more as a catalog of what has and hasnt worked and more context around why.  

Would providing more reasoning/clarity/what-it-does assist with the milestones? Such that as a project maintainer looking to move to incubation from sandbox (for instance) has ample information about how these may be beneficial   to decide if documenting a roadmap is worthwhile for their project, or perhaps including a vision/direction, or neither, works for their needs?

Reply:
I've suggested different language for this point.